### PR TITLE
Add forcemerge threadpools

### PR DESCRIPTION
**Pull request by redcinelli**
## What does this PR do?

This Pr should enable the `force_merge` threadpool to be shipped in the monitoring event.

## Why is it important?

We observe more and more client having the Frozen tier, and in order to have snapshot a`Force_merge` operation have to be done on each shards.
This can prove to be quite the load on the cluster, and being able to monitor it does not seems like a bad idea.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

I created this PR, based off https://github.com/elastic/beats/pull/28479/files#diff-2b535cf7ec876087501fcaae889683e21f2acaab95d2274c01f647a1e1ef4e7c

### CODE REVIEW

**Comment by redcinelli** [comment]
I see that 2 checks are failing with the error `Error: some files are not up-to-date. Run 'make update' then review and commit the changes.`

To fix it I do `mage update`

But it is updating quite a lot of file, I feel a review might be easier before pushing those changes.

```
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   beater/doc.go
	modified:   beater/metricbeat.go
	modified:   docs/fields.asciidoc
	modified:   mb/event.go
	modified:   mb/mb.go
	modified:   mb/module/options.go
	modified:   mb/testing/modules.go
	modified:   module/consul/testing.go
	modified:   module/doc.go
	modified:   module/docker/container/container.go
	modified:   module/docker/cpu/cpu.go
	modified:   module/docker/diskio/diskio.go
	modified:   module/docker/healthcheck/healthcheck.go
	modified:   module/docker/image/image.go
	modified:   module/docker/memory/memory.go
	modified:   module/docker/network/network.go
	modified:   module/docker/network_summary/config.go
	modified:   module/elasticsearch/fields.go
	modified:   module/golang/heap/data.go
	modified:   module/golang/util.go
	modified:   module/jolokia/jmx/config.go
	modified:   module/jolokia/jmx/data.go
	modified:   module/kafka/doc.go
	modified:   module/linux/rapl/rapl.go
	modified:   module/mysql/mysql.go
	modified:   module/postgresql/metricset.go
	modified:   module/postgresql/postgresql.go
	modified:   module/postgresql/testing.go
	modified:   module/redis/info/doc.go
	modified:   module/system/filesystem/doc.go
	modified:   module/system/fsstat/doc.go
	modified:   module/system/process/doc.go
	modified:   module/system/raid/blockinfo/getdev.go
	modified:   module/system/raid/blockinfo/parser.go
	modified:   module/system/service/dbus.go
	modified:   module/system/socket_summary/sockstat_linux.go
	modified:   module/system/socket_summary/sockstat_other.go
	modified:   module/windows/service/defs_service_windows.go
	modified:   module/zookeeper/mntr/mntr.go
	modified:   module/zookeeper/server/server.go
```

**Comment by redcinelli** [comment]
As for back port of the feature.
I feel a PM might want to chip in ? Else I would not mind having it back ported as far as it can ^^

Also, applying the changes done by the `make update` might severely impact the back portability of this pr ^^

**Comment by klacabane** [comment]
The data generated by the elasticsearch metricset can be delivered to two other places:
- [.monitoring-es-mb template](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/resources/monitoring-es-mb.json) template
- [elasticsearch integration](https://github.com/elastic/integrations/tree/main/packages/elasticsearch/data_stream/node_stats)

Could you please add the mappings related to these fields there ?

File: metricbeat/docs/fields.asciidoc [hunk 1]
```diff
@@ -29819,6 +29819,23 @@ alias to: elasticsearch.node.stats.thread_pool.bulk.rejected.count
 
 --
 
+*`node_stats.thread_pool.force_merge.queue`*::
```
Comment (klacabane) [inline metricbeat/docs/fields.asciidoc:29822]: I don't see these aliases defined in the mappings which is probably what is causing the build to fail.
For context, these aliases exist to make Stack Monitoring application querying easier and if we have no plan to use them there we can remove them.

**Comment by redcinelli** [comment]
> The data generated by the elasticsearch metricset can be delivered to two other places:
> 
> * [.monitoring-es-mb template](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/resources/monitoring-es-mb.json) template
> * [elasticsearch integration](https://github.com/elastic/integrations/tree/main/packages/elasticsearch/data_stream/node_stats)
> 
> As a follow up, could you please add the mappings related to these fields there ?

Yes, here is:

- The [Integration repo](https://github.com/elastic/integrations/pull/5508) Pr
- The [Elasticsearch repo](https://github.com/elastic/elasticsearch/pull/94469) (draft)PR

File: metricbeat/docs/fields.asciidoc (hunk 1 above)
Comment (redcinelli) [reply metricbeat/docs/fields.asciidoc]: 🪄  it is gone ^^

Comment (redcinelli) [reply metricbeat/docs/fields.asciidoc]: But I believe this is not what make the build fail.
From the build logs it is `Error: some files are not up-to-date. Run 'make update'` which I am aware of but for review purposes I felt it was better not to include those changes **yet**.

**Comment by klacabane** [comment]
I committed the `mage update` change which only updated the relevant fields snapshot. Not sure why it would modify all those files on your end, do you have any pending local changes ?

**Review by klacabane — APPROVED** [review]

**Pull request merged**